### ISSUE TITLE
variation data tab radio inputs are unchecked after reordering variation

### ISSUE THREAD

**Issue by tamas-web** [woothemes/woocommerce#35109]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I am trying to add a couple of radio buttons to variations data tabs with the `woocommerce_product_after_variable_attributes` hook. So when ever you have a radio option selected, and reorder a specific variation, all radio inputs for that variation get unchecked, whether the variation is opened or closed for editing.

### Expected behavior

All radio inputs should preserve checked values.

### Actual behavior

All radio inputs are unchecked on variation, which is being reordered.
![Screenshot 2022-10-15 204127](https://user-images.githubusercontent.com/15021943/196001754-bccf1ab8-a05f-45f6-a71b-0906e78b1493.png)
![Screenshot 2022-10-15 204154](https://user-images.githubusercontent.com/15021943/196001757-f378c4c5-cef7-4d12-8fd9-28026736b28c.png)

### Steps to reproduce

```
<?php

add_action( 'woocommerce_product_after_variable_attributes', 'bug_product_type_variation', 10, 3 );

function bug_product_type_variation( $loop, $variation_data, $variation ){

	$product = wc_get_product( $variation->ID );

	$opt = $product->get_meta( 'opt' );
	$opt = $opt ? $opt : 'opt1';

	echo '<div class="options">';

		woocommerce_wp_radio( [
			'id'            => 'prefix_opt',
			'name'          => sprintf( 'prefix_opt[%u]', $loop ),
			'label'         => __( 'Options', 'textdomain' ),
			'value'         => $opt,
			'options'       => [
				'opt1' => __( 'Option 1', 'textdomain' ),
				'opt2' => __( 'Option 2', 'textdomain' ),
			],
		] );

		echo '<p class="form-field">
			<label for="prefix_opt1_raw_', $loop, '">', esc_html( 'Option 1', 'textdomain' ), '</label>
			<input type="radio" id="prefix_opt1_raw_', $loop, '" name="', sprintf( 'prefix_opt2[%u]', $loop ), '" value="opt1" ', checked( $opt, 'opt1', false ), '>
		</p>';

		echo '<p class="form-field">
			<label for="prefix_opt2_raw_', $loop, '">', esc_html( 'Option 2', 'textdomain' ), '</label>
			<input type="radio" id="prefix_opt2_raw_', $loop, '" name="', sprintf( 'prefix_opt2[%u]', $loop ), '" value="opt2" ', checked( $opt, 'opt2', false ), '>
		</p>';

		printf( '<label>Is for sure checked?<input type="radio" name="test[%u]" value="on" checked="checked"/></label>', $loop );

		echo '<p class="form-field">
			<label for="prefix_opt3_raw_', $loop, '">', esc_html( 'Option 3', 'textdomain' ), '</label>
			<input type="checkbox" id="prefix_opt3_raw_', $loop, '" name="', sprintf( 'prefix_opt3[%u]', $loop ), '" value="opt2" ', checked( $opt, 'opt2', false ), '>
		</p>';

	echo '</div>';

}

```

### WordPress Environment ###

WordPress address (URL): http://localhost/wp
Site address (URL): http://localhost/wp
WC Version: 7.0.0
REST API Version: ✔ 7.0.0
WC Blocks Version: ✔ 8.5.1
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6
PHP Version: 8.1.6
PHP Post Max Size: 40 MB
PHP Time Limit: 120
PHP Max Input Vars: 1000
cURL Version: 7.77.0
OpenSSL/1.1.1n

SUHOSIN Installed: –
MySQL Version: 10.4.24-MariaDB
Max Upload Size: 40 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.0.0
WC Database Prefix: wp_
Total Database Size: 4.81MB
Database Data Size: 3.37MB
Database Index Size: 1.44MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.44MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
product: 3
product_variation: 3

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 7.0.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty Two Child
Version: 1.0.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Twenty Twenty-Two
Parent Theme Version: 1.2
Parent Theme Author URL: https://wordpress.org/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-10-16 17:29:08 +00:00
Options: ✔
Notes: 36
Onboarding: skipped

### Action Scheduler ###

Complete: 16
Oldest: 2022-10-15 17:29:42 +0000
Newest: 2022-10-15 17:32:42 +0000

Pending: 4
Oldest: -0001-11-30 00:00:00 +0000
Newest: 2022-10-16 17:29:42 +0000

### Status report information ###

Generated at: 2022-10-15 17:33:09 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by tamas-web** [comment]
I am not requesting any support assistance. I provided a code sample so you can check the issue which I'm refering to.
I'll try to explain the bug again. When ever you have a radio input in a variation, and you try to reorder that variation, all radio input settings for that variation loose their checked status. They become unchecked.

**Comment by tamas-web** [comment]
So are you just going to ignore this bug and mark it as support request? Have you even tested this issue?
Also I have found another bug, or at least it seems like a bug. When you try to reorder variation by clicking and setting a menu order, after a couple of reorders it's not reordering properly anymore. I have provided a video so you don't have to use the provided code.
Again, radio inputs are unchecked upon variation reorder.  

https://user-images.githubusercontent.com/15021943/200162851-f195f179-be54-472a-a359-1a51e8e35ae4.mp4
